src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java

```diff
@@ -218,7 +218,10 @@ public void beforeCommand(CommandEnvironment env) throws AbruptExitException {
               (execChannel != null ? execChannel : cacheChannel),
               credentials,
               remoteOptions.remoteTimeout,
-              retrier);
+              retrier,
+              (execChannel != null ?
+                  TracingMetadataUtils.newExecHeadersInterceptor(remoteOptions) :
```

Comment:
I find it slightly odd that this has different options for caching and execution. If someone is asking Bazel to do one or the other, they can also provide different options, no?

(I realize that's a pre-existing condition, so feel free to ignore this comment.)

Reply:
Updated to run the checks twice if the endpoints differ.